opentelemetry-api/src/opentelemetry/metrics/__init__.py

```diff
@@ -11,3 +11,398 @@
+measurements, as well as metrics with known aggregation and labels.
+
+The :class:`.Meter` class is used to construct :class:`.Measure` s to
+record raw measurements and :class:`.Metric` s to record metrics with
+predefined aggregation.
+
+See the `metrics api`_ spec for terminology and context clarification.
+
+.. _metrics api:
+    https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-metrics.md
+
+
+"""
+
+import abc
+import typing
+
+from opentelemetry.metrics.label_key import LabelKey
+from opentelemetry.metrics.label_value import LabelValue
+from opentelemetry.metrics.time_series import CounterTimeSeries
+from opentelemetry.metrics.time_series import GaugeTimeSeries
+from opentelemetry.resources import Resource
+from opentelemetry.trace import SpanContext
+
+
+class Meter:
+    """An interface to allow the recording of measurements and metrics.
+
+    :class:`.Measurement` s are used for recording raw values, in which the
+    aggregation and labels for the exported metric are defered. This should be
+    used to record measurements like "server_latency" or "received_bytes",
+    where the value of interest is the recorded value itself.
+
+    :class:`.Metric` s are used for recording pre-defined aggregation, or
+    already aggregated data. This should be used to report metrics like
+    cpu/memory usage, in which the type of aggregation is already defined, or
+    simple metrics like "queue_length".
+    """
+
+    def create_float_counter(self,
+                             name: str,
+                             description: str,
+                             unit: str,
+                             label_keys: typing.List['LabelKey'],
+                             constant_labels: \
+                             typing.Dict['LabelKey', 'LabelValue'] = None,
+                             component: str = None,
+                             resource: 'Resource' = None
+                             ) -> 'CounterFloat':
+        """Creates a counter type metric that contains float values.
+
+        Args:
+            name: The name of the counter.
+            description: Human readable description of the metric.
+            unit: Unit of the metric values.
+            label_keys: list of keys for the labels with dynamic values. \
+            Order of the list is important as the same order MUST be used \
+            on recording when suppling values for these labels
+            constant_labels: A map of constant labels that will be used for \
+            all of the TimeSeries created from the Metric. \
+            component: The name of the component that reports this metric.
+            Resource: Sets the :class:`.Resource` associated with this metric.
+
+        Returns: A new :class:`.CounterFloat`
+        """
+
+    def create_int_counter(self,
+                           name: str,
+                           description: str,
+                           unit: str,
```

Comment:
random question: why is "unit" it's own field when it could be identified as a label_key?

I feel like not everything measured needs a unit to help understand the value. And even if it did, I don't believe many metric storage systems have the unit as a field they store. 

For example, I believe graphite or prometheus would store the "unit" key as a tag or a part of the metric name.